### PR TITLE
Move RSA keys to their own file and use a pem file for them

### PR DESCRIPTION
**Pull request by cubeee**
RSA keys are sensitive information that should not be stored in git repositories.
This commit moves them to their own file, adds it to .gitignore and provides a template file.

### CODE REVIEW

**Comment by garyttierney** [comment]
I think storing the key PKCS#8 would be a better option since it's something that NSS, OpenSSL, and the JDK can all work with.

**Comment by Major-** [comment]
Originally this was for ease-of-setup (the public counterparts are in all the clients I made available to people) but in general it would be loads nicer to have a proper ./configure script that generates the keys etc locally

**Comment by garyttierney** [comment]
@cubeee status on this?

**Comment by cubeee** [comment]
@garyttierney done unless you had something in mind

**Comment by garyttierney** [comment]
@cubeee LGTM, just needs rebase.

File: net/src/main/org/apollo/net/NetworkConstants.java [hunk 1]
```diff
@@ -76,6 +74,20 @@
...
+			PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(pem.getContent());
+
+			Security.addProvider(new BouncyCastleProvider());
+			KeyFactory factory = KeyFactory.getInstance("RSA", "BC");
+
+			RSAPrivateKey privateKey = (RSAPrivateKey) factory.generatePrivate(keySpec);
+			RSA_MODULUS = privateKey.getModulus();
+			RSA_EXPONENT = privateKey.getPrivateExponent();
+		} catch (Exception exception) {
+			throw new ExceptionInInitializerError(new IOException("Error parsing id_rsa", exception));
```
Comment (garyttierney) [inline net/src/main/org/apollo/net/NetworkConstants.java]: Confusing error message, id_rsa doesn't exist

**Pull request merged**